migration/src/audio.rs

```diff
@@ -1,38 +1,174 @@
+        self.0.keys().next().unwrap().clone()
+    }
+}
+
+/// Represents info associated with an object from the DRS.
+/// 
+/// # Examples:
+/// The HTTP Response
+/// ```text
+/// {
+///     "pid": "neu:br86b3634",
+///     "parent": "neu:n009w288x",
+///     "thumbnails": [
+///         "https://repository.library.northeastern.edu/downloads/neu:4f1738386?datastream_id=thumbnail_1",
+///         "https://repository.library.northeastern.edu/downloads/neu:4f1738386?datastream_id=thumbnail_2",
+///         "https://repository.library.northeastern.edu/downloads/neu:4f1738386?datastream_id=thumbnail_3",
+///         "https://repository.library.northeastern.edu/downloads/neu:4f1738386?datastream_id=thumbnail_4",
+///         "https://repository.library.northeastern.edu/downloads/neu:4f1738386?datastream_id=thumbnail_5"
+///         ],
+///     "canonical_object": {
+///         "https://repository.library.northeastern.edu/downloads/neu:4f173836n?datastream_id=content": "Audio File"
+///     },
+/// }
+/// ```
+/// becomes
+/// ```
+/// DrsRes {
+///     pid = "neu:br86b3634",
+///     parent = "neu:n009w288x",
+///     thumbnails = vec::new([
+///         "https://repository.library.northeastern.edu/downloads/neu:4f1738386?datastream_id=thumbnail_1",
+///         "https://repository.library.northeastern.edu/downloads/neu:4f1738386?datastream_id=thumbnail_2",
+///         "https://repository.library.northeastern.edu/downloads/neu:4f1738386?datastream_id=thumbnail_3",
+///         "https://repository.library.northeastern.edu/downloads/neu:4f1738386?datastream_id=thumbnail_4",
+///         "https://repository.library.northeastern.edu/downloads/neu:4f1738386?datastream_id=thumbnail_5"
+///     ])
+///     canonical_object: ComplexDrsObject(["https://repository.library.northeastern.edu/downloads/neu:4f173836n?datastream_id=content": "Audio File"])
+/// }
+/// ```
 #[derive(Serialize, Deserialize, Clone, Debug)]
 struct DrsRes {
+    /// The PID for a DRS Object. Takes the form "neu:XXXXXXXXX"
     pid: DrsId,
-    // breadcrumbs: ComplexDrsObject,
+    /// The PID for the parent collection of this object.
     parent: DrsId,
+    /// A list of URLs leading to this object's thumbnail image.
+    /// 
+    /// Each URL represents a different size of image.
     thumbnails: Vec<String>,
+    /// The primary data associated with this object and its data type.
     canonical_object: ComplexDrsObject,
-    // content_objects: ComplexDrsObject,
 }
 
 impl DrsRes {
-    // TODO: flatten information held in ComplexDrsObjects
+    /// Creates a new DRS resource
+    ///
+    /// # Errors
+    /// Fails if a connection with the DRS can not be established in a set number of tries
+    /// or if the response body cannot be deserialized properly.
     pub async fn new(client: &Client, drs_id: &str) -> Result<Self, anyhow::Error> {
         let drs = "https://repository.library.northeastern.edu/api/v1/files/";
         let mut tries = 0;
+
         Ok(loop {
```

Comment:
when would this loop be broken out of? reading the rust documentation for a loop, it looks like this is an infinite loop until a `break` is called

Reply:
When the client returns an ok result or tries exceed 7. See lines 180-183.